Marlin/src/lcd/menu/menu_tramming.cpp

```diff
@@ -50,32 +52,32 @@ static bool probe_single_point() {
   do_blocking_move_to_z(TERN(BLTOUCH, Z_CLEARANCE_DEPLOY_PROBE, Z_CLEARANCE_BETWEEN_PROBES));
   // Stow after each point with BLTouch "HIGH SPEED" mode for push-pin safety
   const float z_probed_height = probe.probe_at_point(tramming_points[tram_index], TERN(BLTOUCH_HS_MODE, PROBE_PT_STOW, PROBE_PT_RAISE), 0, true);
-  DEBUG_ECHOLNPAIR("probe_single_point: ", z_probed_height, "mm");
   z_measured[tram_index] = z_probed_height;
+  if (reference_index < 0) reference_index = tram_index;
   move_to_tramming_wait_pos();
 
-  return !isnan(z_probed_height);
+  DEBUG_ECHOLNPAIR("probe_single_point(", tram_index, ") = ", z_probed_height, "mm");
+  return (z_isvalid[tram_index] = !isnan(z_probed_height));
 }
 
-static void _menu_single_probe(const uint8_t point) {
-  tram_index = point;
-  DEBUG_ECHOLNPAIR("Screen: single probe screen Arg:", point);
+static void _menu_single_probe() {
+  tram_index = MenuItemBase::itemIndex;
+  DEBUG_ECHOLNPAIR("Screen: single probe screen Arg:", tram_index);
   START_MENU();
   STATIC_ITEM(MSG_BED_TRAMMING, SS_LEFT);
-  STATIC_ITEM(MSG_LAST_VALUE_SP, SS_LEFT, ftostr42_52(z_measured[0] - z_measured[point])); // Print diff
+  STATIC_ITEM(MSG_LAST_VALUE_SP, SS_LEFT, z_isvalid[tram_index] ? ftostr42_52(z_measured[reference_index] - z_measured[tram_index]) : "---");
   ACTION_ITEM(MSG_UBL_BC_INSERT2, []{ if (probe_single_point()) ui.refresh(); });
-  ACTION_ITEM(MSG_BUTTON_DONE, []{ ui.goto_previous_screen(); }); // Back
+  ACTION_ITEM(MSG_BUTTON_DONE, []{ ui.goto_previous_screen(); });
   END_MENU();
 }
 
 static void tramming_wizard_menu() {
-  DEBUG_ECHOLNPAIR("Screen: tramming_wizard_menu");
   START_MENU();
   STATIC_ITEM(MSG_SELECT_ORIGIN);
 
   // Draw a menu item for each tramming point
   LOOP_L_N(i, G35_PROBE_COUNT)
-    SUBMENU_N_P(i, (char*)pgm_read_ptr(&tramming_point_name[i]), []{ _menu_single_probe(MenuItemBase::itemIndex); });
+    SUBMENU_N_P(i, (char*)pgm_read_ptr(&tramming_point_name[i]), _menu_single_probe);
```

Comment:
I'm surprised that the value of `i` is not available to be passed in `MenuItemBase::itemIndex` as this construct is used in many places. For example…
```cpp
      LOOP_L_N(n, EXTRUDERS)
        EDIT_ITEM_N(int3, n, MSG_FLOW_N, &planner.flow_percentage[n], 10, 999, []{ planner.refresh_e_factor(MenuItemBase::itemIndex); });
```
I see that you decided to follow the same pattern as `CANCEL_OBJECT`, picking up the index in the called handler above. That should work. However, I would definitely prefer to standardize and figure out a more elegant way to pass the index directly.

Reply:
I think the key difference is that in the working example it is passing a static variable and not a local.

I didn’t spend too much time fighting it. If you can get it to build that’s great, but it wouldn’t compile how it was.

Removing the argument from that function call actually decreased FLASH usage by a few dozen bytes as well.

Reply:
It was actually failing on the single point menu, where it was trying to add a menu item that was a local function argument.

Reply:
You might be able to just explicitly capture that by value. I know I’ve hit limitations in the past on using lambdas without the full STL available, but I can’t remember what the limitations were.

Reply:
Right, because `MenuItemBase::itemIndex` is static and global.